misc/python/materialize/cli/ci_logged_errors_detect.py

```diff
@@ -260,6 +264,28 @@ def get_error_logs(log_files: list[str]) -> list[ErrorLog]:
                 ):
                     continue
                 error_logs.append(ErrorLog(match.group(0), log_file))
+            open_panics = {}
+            for line in iter(data.readline, b""):
+                line = line.rstrip(b"\n")
+                if match := PANIC_START_RE.match(line):
+                    service = match.group("service")
+                    assert (
+                        service not in open_panics
+                    ), f"Two panics of same service {service} interleaving: {line}"
+                    open_panics[service] = line
+                elif open_panics:
+                    if match := PANIC_CONTINUED_RE.match(line):
+                        if panic_start := open_panics.get(match.group("service")):
```

Comment:
Add an else-case with an assertion?
The service may no longer be in `open_panics` if two panics of the same service interleave?

Reply:
There is intentionally no else-case here, and an assertion in it would be wrong. The `PANIC_CONTINUED_RE` will match every line that follows, and its main purpose is to split them into service and message. I renamed it to `SERVICES_LOG_LINE_RE` now to make this clearer. Also added a comment.